java/org/apache/juli/FileHandler.java

```diff
@@ -585,4 +517,34 @@ private String obtainDateFromPath(Path path) {
             return null;
         }
     }
+
+
+    static class AsyncTaskExecutor {
+
+        private Thread thread;
+        private BlockingQueue<Runnable> tasks;
+
+        AsyncTaskExecutor() {
+            this.tasks = new LinkedBlockingQueue<>();
+            this.thread = new Thread(() -> {
+                try {
+                    this.tasks.take().run();
+                } catch (Exception e) {
+                    System.err.println("take task failed");
+                    e.printStackTrace();
+                }
+            });
+            this.thread.setName("tomcat-async-delete-file");
+            this.thread.start();
```

Comment:
The thread should be set as a daemon, or the class should be a lifecycle listener and stop the thread when Tomcat shuts down.